### ISSUE TITLE
Standalone in linux cannot load MATPLOTLIB

### ISSUE THREAD

**Issue by LuisTbx** [Nuitka/Nuitka#883]
* Nuitka version, full Python version and Platform (Windows, OSX, Linux ...)
pip 20.2.4
Python 3.6.9
Linux, ubuntu 18.04
Nuitka 0.6.9.6

* How did you install Nuitka and Python (pip, anaconda, deb, rpm, from source,
  what is a virtualenv ...), this is very important usually.

Using Nuitka 0.6.9.6 from pip
Installed inside a virtualenv.

* If possible please supply a [Short, Self Contained, Correct, Example](http://sscce.org/)
  that demonstrates the issue i.e a small piece of code which reproduces
  the issue and can be run with out any other (or as few as possible)
  external dependencies.
one liner test:

from matplotlib import pyplot as plt

similar to: [https://github.com/Nuitka/Nuitka/issues/598]

Compiling with:
python -m nuitka --standalone --follow-imports --plugin-enable=pylint-warnings --plugin-enable=numpy test.py

Error

testMATPLOT/test.dist$ ./test
Traceback (most recent call last):
File "testMATPLOT/test.dist/test.py", line 1, in <module>
File "<frozen importlib._bootstrap>", line 971, in _find_and_load
File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
File "testMATPLOT/test.dist/matplotlib/pyplot.py", line 2429, in <module matplotlib.pyplot>
File "testMATPLOT/test.dist/matplotlib/pyplot.py", line 95, in _copy_docstring_and_deprecators
KeyError: 'func'

**Comment by kayhayen** [comment]
I found the code that causes it:

```
                kwargs = {
                    k: v.cell_contents
                    for k, v in zip(code.co_freevars, method.__closure__)}
                assert kwargs["func"] is method.__wrapped__
                kwargs.pop("func")
                decorators.append(decorator_maker(**kwargs))
```

So this is a compatibility issue. I don't think we set `co_freevars` currently. I think the `__closure__` attribute is good though, but `zip` doesn't notice, and takes the shortest list. It wants to ignore the closure variable `func` it seems.

**Comment by kayhayen** [comment]
I am hesitant to add this feature without first making the scalability enhancement to generate these code objects from constant data rather than from C code, needs some thought, as I really don't want to delay 0.6.10 any further, but this seems to be an important incompatibility.

**Comment by kayhayen** [comment]
Seems, the `co_varnames` is not supposed to contain names from `co_freevars`. Nuitka put the closure variable in the former, and will now put it in the later, so this addresses 2 compatibility problems at once.

To go easy on my premature optimization mind, I enhanced the code generation for code objects such that it's not going to put too much extra burden. I am running internal tests, and there are a few findings, that I will need to address first.

**Comment by kayhayen** [comment]
I need to keep the incompatibility of `co_varnames` including values of `co_freevars`, it's too deeply tied into the code, of compiled functions expecting all names to be there. However this has not yet been observed as a problem, so that's harmless likely.

**Comment by kayhayen** [comment]
I justed pushed the fix to factory branch, this should help.

**Comment by LuisTbx** [comment]
Thanks for the fast answer Kay! 

Cloned the GIT repo, checked out the factory branch. Created a new virtualenv, installed matplotlib, cython and numpy. Ran setup.py build, setup.py install. Compiled a basic example and.... it all works! 

I believe this issue can be closed now.

**Comment by kayhayen** [comment]
We keep issus that affect the stable version open until release.

**Comment by kayhayen** [comment]
Released as part of 0.6.10
